src/Umbraco.Core/Composing/LightInject/LightInjectContainer.cs

```diff
@@ -56,9 +57,24 @@ protected static ServiceContainer CreateServiceContainer()
             // it is an index property. which means that eg protected or internal setters are OK.
             //Container.EnableAnnotatedPropertyInjection();
 
-            // ensure that we do *not* scan assemblies
-            // we explicitly RegisterFrom our own composition roots and don't want them scanned
-            container.AssemblyScanner = new AssemblyScanner(/*container.AssemblyScanner*/);
+            // ensure that we do *not* scan our assemblies
+            // but do not prevent users from doing so if they so desire.
+            bool PreventScanForUmbracoCompositionRootsPredicate(Type type, string s)
+            {
+                // Tests expect the other registered fallback to run, see RegisterAuto below.
+                if (!type.IsInterface)
+                    return false;
+
+                // The fallback factory method only applies to types in Umbraco namespaces
+                return type.FullName?.StartsWith("Umbraco", StringComparison.OrdinalIgnoreCase) ?? false;
```

Comment:
Is this too restrictive? Some user's may have assembly names that start with Umbraco (i.e. `UmbracoIdentity`). Maybe it should be "Umbraco." ?

Reply:
Agreed 100%